base/pkg/entry.jl

```diff
@@ -290,6 +290,52 @@ function free(pkgs)
     end
 end
 
+function freeable(io::IO = STDOUT)
+    freelist = Any[]
+    cd(Dir.path()) do
+        instd = Read.installed()
+        first = true
+        k, v = collect(keys(instd)), collect(values(instd))
+        p = sortperm(k)
+        k = k[p]
+        v = v[p]
+        for (pkg,status) in zip(k,v)
+            ver, fix = status
+            if fix
+                repo = GitRepo(pkg)
```

Comment:
use `LibGit2.with` function to invoke `do` pattern to guaranty resource deallocation. 
```julia
LibGit2.with(GitRepo(pkg)) do repo
# working with `repo`
end
```